src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -20,6 +21,112 @@
 ReflowSequenceType = List[Union[ReflowBlock, ReflowPoint]]
 
 
+@dataclass(frozen=True)
+class _RebreakSpan:
+    """A location within a sequence to consider rebreaking."""
+
+    target: BaseSegment
+    start_idx: int
+    end_idx: int
+    line_position: str
+
+
+@dataclass(frozen=True)
+class _RebreakLocation:
+    """A location within a sequence to rebreak, with metadata."""
+
+    target: BaseSegment
+    prev_code_pt_idx: int
+    prev_nl_idx: int
+    prev_point_idx: int
+    next_point_idx: int
+    next_nl_idx: int
+    next_code_pt_idx: int
+    line_position: str
+
+    @classmethod
+    def from_span(cls, span: _RebreakSpan, elements: ReflowSequenceType):
+        """Expand a span to a location."""
+        # First get the next newline.
+        prev_point_idx = span.start_idx - 1
+        next_point_idx = span.end_idx + 1
+
+        prev_nl_idx = prev_point_idx
+        next_nl_idx = next_point_idx
+        # We hop in 2s because we're checking two ahead.
+        while (
```

Comment:
Could these two `while` loop be rewritten as `for` loops? Something like:
```
for prev_nl_idx in range(prev_point_idx, 2, -2):
```

For reference:
```
>>> list(range(13, 2, -2))
[13, 11, 9, 7, 5, 3]
```

Context: `for` loops are easier to read and it's clearer that they will definitely terminate.

Reply:
Good idea.